lib/plugins/aws/provider/awsProvider.js

```diff
@@ -13,13 +13,17 @@ const fs = require('fs');
 const objectHash = require('object-hash');
 const PromiseQueue = require('promise-queue');
 const getS3EndpointForRegion = require('../utils/getS3EndpointForRegion');
+const readline = require('readline');
 
 const constants = {
   providerName: 'aws',
 };
 
 PromiseQueue.configure(BbPromise.Promise);
 
+// Store credentials in this variable to avoid creating them several times (messes up MFA).
+let cachedCredentials = null;
```

Comment:
Move to `this.cachedCredentials` in the `AwsProvider`'s constructor.

Reply:
I think [x] Allow edits from maintainers is enabled, should I enable something more to make it work?

Reply:
Doh. You're right, I'm just not used to pushing to other peoples repos. Got it now.